File: Base 7 - Leetcode 504/Base 7 - Leetcode 504.java
public class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        int originalNum = num;
        num = Math.abs(num);
        StringBuilder remainders = new StringBuilder();

        while (num > 0) {
            int remainder = num % 7;
            remainders.append(remainder);
            num /= 7;
        }

        if (originalNum < 0) {
            remainders.append('-');
        }

        return remainders.reverse().toString();
    }
}

File: Base 7 - Leetcode 504/Base 7 - Leetcode 504.js
var convertToBase7 = function(num) {
    if (num === 0) {
        return '0';
    }

    var originalNum = num;
    num = Math.abs(num);
    var remainders = [];

    while (num > 0) {
        var remainder = num % 7;
        remainders.push(String(remainder));
        num = Math.floor(num / 7);
    }

    if (originalNum < 0) {
        remainders.push('-');
    }

    remainders.reverse();
    return remainders.join('');
};
